bd-logger/src/builder.rs

```diff
@@ -297,39 +299,12 @@ impl LoggerBuilder {
 
       bd_client_common::error::UnexpectedErrorHandler::register_stats(&scope);
 
-      let session_strategy = self.params.session_strategy;
-
-      // By running it before we start all the other components, we ensure that the crash is
-      // processed before cached configuration is loaded, allowing us to pass a fixed set of logs
-      // to the async buffer that it can emit once it transitions into the configured mode,
-      // emitting these logs before any logs emitted by the application while we were starting
-      // up.
-      let crash_logs = crash_monitor
-        .process_new_reports()
-        .await
-        .into_iter()
-        .map(|crash_log| Log {
-          log_level: log_level::ERROR,
-          log_type: LogType::Lifecycle,
-          message: crash_log.message,
-          fields: crash_log.fields,
-          matching_fields: [].into(),
-          session_id: session_strategy
-            .previous_process_session_id()
-            .unwrap_or_else(|| session_strategy.session_id()),
-          occurred_at: crash_log.timestamp,
-          // Always capture the session when we process a crash log.
-          // TODO(snowp): Ideally we should include information like the report and client side
-          // grouping here to help make smarter decisions during intent negotiation.
-          capture_session: Some("crash_handler".to_string()),
-        })
-        .collect();
-
       try_join!(
```

Comment:
This is the top level future that basically never exits. I don't think this change and the related change in running the logger runtime is what you want. Basically just create the monitor above and return it, unless I'm missing something? then you would communicate with it using the channel.